include/envoy/grpc/async_client.h

```diff
@@ -195,7 +247,24 @@ class AsyncClient {
   virtual AsyncRequest* send(const Protobuf::MethodDescriptor& service_method,
                              const Protobuf::Message& request, AsyncRequestCallbacks& callbacks,
                              Tracing::Span& parent_span,
-                             const absl::optional<std::chrono::milliseconds>& timeout) PURE;
+                             const absl::optional<std::chrono::milliseconds>& timeout);
+
+  /**
+   * Start a gRPC unary RPC asynchronously.
+   * @param service_full_name full name of the service (i.e. service_method.service()->full_name()).
+   * @param method_name name of the method (i.e. service_method.name()).
+   * @param request serialized message.
+   * @param callbacks the callbacks to be notified of RPC status.
+   * @param parent_span the current parent tracing context.
+   * @param timeout supplies the request timeout.
+   * @return a request handle or nullptr if no request could be started. NOTE: In this case
+   *         onFailure() has already been called inline. The client owns the request and the
+   *         handle should just be used to cancel.
+   */
+  virtual AsyncRequest* sendRaw(absl::string_view service_full_name, absl::string_view method_name,
```

Comment:
Yeah, I reckon we should have the gRPC implementations just implement the raw variants, and we can layer a wrapper layer to do the rest on top of this. That will shrink the interface a bit.

Reply:
Let's discuss this.  The implementation only works with the Raw version except for Protobuf::MethodDescriptor which it must convert to the service_full_name and method_name.  This conversion accounts for start/startRaw ->send/sendRaw.   For clarity the raw forms also take the raw versions of the callbacks.  However the implementations do not know about the non-Raw forms of the callbacks.  So in theory we could just overload send()/start() to do the Protobuf::MethodDescriptor conversion and have a single for for the callbacks, but then we would still want the other forms of callbacks although we could move them into the source/... tree.  However I am not sure how that would work with the client code and it would complicate the tests.  Right now we can mock all the versions and do the tests at the most natural level (typically not Raw).  Let's discuss it.

Reply:
Updated to the raw only API.